### PR TITLE
Pick up default namespace from config

### LINKED ISSUE #7: pick up default namespace from .kube/config instead of hardcoding it in kn

### PR DESCRIPTION
**Pull request by tzununbekov**
Namespace parameter moved to root persistent flags. Function `setNamespace()` sets namespace from config context on CLI initialization. Resolves #7

### CODE REVIEW

**Comment by tzununbekov** [comment]
this PR includes changes from #27, so #27 should be merged first I think

**Comment by cppforlife** [comment]
@tzununbekov we should use client-go functionality for this. see knctl for example: https://github.com/cppforlife/knctl/blob/master/pkg/knctl/cmd/core/config_factory.go#L77

**Comment by tzununbekov** [comment]
Looks like I missed this client-go function, thanks for pointing out. I'll update PR

**Comment by sebgoa** [comment]
@tzununbekov can you update this PR with the client-go function

**Comment by cppforlife** [comment]
@tzununbekov still up for making a change to use client-go?

**Comment by tzununbekov** [comment]
nice, client now has tests!
yes, I'm still up to make these changes though it's turned out to be a bit more code than I expected :slightly_smiling_face: 
I think I'll be able to push it tomorrow (eastern hemisphere) and I also see several more issues which I can help with

**Comment by bbrowning** [comment]
Looks like the feedback to use client-go was addressed. @tzununbekov is this ready for review, or were those other issues you mentioned something that would be part of this PR?

**Comment by tzununbekov** [comment]
> Looks like the feedback to use client-go was addressed. @tzununbekov is this ready for review, or were those other issues you mentioned something that would be part of this PR?

@bbrowning this one is ready - I will submit new PRs for other issues

**Comment by bbrowning** [comment]
I pinged Navid on Slack about how to use Prow commands instead of the GitHub UI to indicate "looks good to me". But, since it's late his local time, I'll nudge this one along. I pulled the changed locally, built the updated `kn` CLI, and verified that the default namespace in my kubeconfig gets used as well as that I can manually override that with the global namespace flag.

/lgtm

File: pkg/kn/commands/root.go [hunk 1]
```diff
@@ -78,11 +82,16 @@ Eventing: Manage event subscriptions and channels. Connect up event sources.`,
 func InitializeConfig() {
 	cobra.OnInitialize(initConfig)
 	cobra.OnInitialize(initKubeConfig)
-
+	cobra.OnInitialize(setNamespace)
 }
 
 func initKubeConfig() {
```
Comment (evankanderson) [inline pkg/kn/commands/root.go]: It would be nice if there were a generic "initKubeConfig" method from the Kubernetes/kubectl libraries.

Add a TODO?

Comment (evankanderson) [reply pkg/kn/commands/root.go]: I think `clientcmd` will do a lot of this resolution if you don't feed it a kubeCfg file. Not sure if empty string argument will trigger its default resolution, or if you need to use a different method.

File: pkg/kn/commands/root.go [hunk 2]
```diff
@@ -129,3 +138,21 @@ func GetConfig() (serving.ServingV1alpha1Interface, error) {
...
+		return
+	}
+	namespace = "default"
+	config := clientConfig(kubeCfgFile)
+	if ns, _, err := config.Namespace(); err == nil {
+		namespace = ns
+	}
+}
+
+func clientConfig(path string) clientcmd.ClientConfig {
```
Comment (cppforlife) [inline pkg/kn/commands/root.go]: here is a better version i cooked up for [kapp](https://github.com/k14s/kapp): https://github.com/k14s/kapp/blob/master/pkg/kapp/cmd/core/config_factory.go#L71-L82. it properly deals with multi-value KUBECONFIG values and picks up default env vars based on cmdclient config. (need to update knctl to do so as well...)

**Comment by tzununbekov** [comment]
@evankanderson @sixolet If I understand your comments correctly, we should try to use kubectl methods where it's possible, so I pushed update which makes kn use kubectl config flags. If this makes sense, we should probably move namespace parameter from persistent flags to `KnParams` or somewhere else. Otherwise I'd bring everything back and borrow config handling from https://github.com/k14s/kapp

**Pull request closed without merging**